numpy/core/fromnumeric.py

```diff
@@ -2790,6 +2790,8 @@ def mean(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
     Returns the average of the array elements.  The average is taken over
     the flattened array by default, otherwise over the specified axis.
     `float64` intermediate and return values are used for integer inputs.
+    `float32` intermediate are used for `float16` inputs for extra
```

Comment:
Since `float16` does nothing unexpected except give and accurate result, I would omit the explanation here and put an extended explanation in the `Notes` section. For bools and ints, something like
```
If ``a`` is of integer or boolean type, dtype defaults to `float64`.
```
However, this is already mentioned in the dtype documentation.